<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1152</width>
    <height>784</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Json Manager v0.3</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>../__main__.svg</normaloff>../__main__.svg</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="mainWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>5</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="leftPanel" native="true">
      <property name="maximumSize">
       <size>
        <width>250</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBoxApiGemini">
         <property name="title">
          <string>Gemini API</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_10">
          <item>
           <widget class="QLabel" name="labelApiKey">
            <property name="text">
             <string>Api Key:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditApiKey">
            <property name="text">
             <string/>
            </property>
            <property name="maxLength">
             <number>39</number>
            </property>
            <property name="echoMode">
             <enum>QLineEdit::PasswordEchoOnEdit</enum>
            </property>
            <property name="placeholderText">
             <string>Enter gemini Api Key here</string>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonReloadModel">
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>Reload Model</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxThemes"/>
       </item>
       <item>
        <widget class="QFontComboBox" name="fontComboBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="editable">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <spacer name="horizontalSpacer_5">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>175</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="centralLayout">
      <item>
       <layout class="QHBoxLayout" name="superiorButtonArea">
        <item>
         <widget class="QPushButton" name="leftHideButton">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>&lt;&lt;&lt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressBarTotal">
          <property name="value">
           <number>0</number>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="rightHideButton">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>&gt;&gt;&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTabWidget" name="centralTabWidget">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="tabPosition">
         <enum>QTabWidget::North</enum>
        </property>
        <property name="tabShape">
         <enum>QTabWidget::Rounded</enum>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <property name="elideMode">
         <enum>Qt::ElideNone</enum>
        </property>
        <property name="documentMode">
         <bool>false</bool>
        </property>
        <property name="movable">
         <bool>false</bool>
        </property>
        <widget class="QWidget" name="tabObjetive">
         <attribute name="title">
          <string>Json objetive</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QTextEdit" name="jsonObjetiveText">
            <property name="html">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7.8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="tabStopWidth">
             <number>40</number>
            </property>
            <property name="acceptRichText">
             <bool>false</bool>
            </property>
            <property name="placeholderText">
             <string>Write the Original Json skeleton Here</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="loadJsonLayout">
            <item>
             <widget class="QLineEdit" name="loadJsonObjetiveRoute"/>
            </item>
            <item>
             <widget class="QPushButton" name="loadJsonObjetiveButton">
              <property name="text">
               <string>Load File</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabTTS">
         <attribute name="title">
          <string>Info to Extract</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="DragDropTabWidget" name="tabInfoToExtract">
            <property name="currentIndex">
             <number>-1</number>
            </property>
            <property name="documentMode">
             <bool>true</bool>
            </property>
            <property name="tabsClosable">
             <bool>true</bool>
            </property>
            <property name="movable">
             <bool>true</bool>
            </property>
            <property name="tabBarAutoHide">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <property name="spacing">
             <number>2</number>
            </property>
            <item>
             <widget class="QPushButton" name="buttonAddTabInfo">
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>Add Tabs</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="buttonSaveAllTabs">
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>Save All Tabs</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="buttonDeleteAllTabs">
              <property name="maximumSize">
               <size>
                <width>120</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Del All Tabs</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabPreprompt">
         <attribute name="title">
          <string>Pre-Prompt(Optional)</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:16pt; font-weight:600; color:#aa0000;&quot;&gt;Warning&lt;/span&gt;&lt;span style=&quot; font-size:16pt; font-weight:600;&quot;&gt;: Only change this if you know what you're doing!&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="textPrePrompt">
            <property name="html">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7.8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;You must fill in the 'json_objective' file with the information provided in 'text_to_extract'. If any information is missing, write it as a null value, but if has a Default value use it. Do not include 'explanation' keys in the JSON, as your sole purpose is to write the filled JSON, and no information can be missing.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="tabStopWidth">
             <number>40</number>
            </property>
            <property name="acceptRichText">
             <bool>false</bool>
            </property>
            <property name="placeholderText">
             <string>This is a preprompt, restart application to restart original prompt.</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabResponse">
         <attribute name="title">
          <string>Last response</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <widget class="QTextEdit" name="textResponse">
            <property name="tabStopWidth">
             <number>40</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="progressBarResponse">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="textVisible">
             <bool>true</bool>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
            <property name="textDirection">
             <enum>QProgressBar::TopToBottom</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="buttonStartProcess">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="text">
            <string>Start Process</string>
           </property>
           <property name="checkable">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QWidget" name="rightPanel" native="true">
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QTextEdit" name="logTextEdit">
         <property name="inputMethodHints">
          <set>Qt::ImhMultiLine</set>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="tabStopWidth">
          <number>40</number>
         </property>
         <property name="placeholderText">
          <string>This is the log file...</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1152</width>
     <height>26</height>
    </rect>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DragDropTabWidget</class>
   <extends>QTabWidget</extends>
   <header>defs/promotions</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>jsonObjetiveText</tabstop>
  <tabstop>loadJsonObjetiveRoute</tabstop>
  <tabstop>loadJsonObjetiveButton</tabstop>
  <tabstop>rightHideButton</tabstop>
  <tabstop>leftHideButton</tabstop>
  <tabstop>centralTabWidget</tabstop>
  <tabstop>textPrePrompt</tabstop>
  <tabstop>textResponse</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
